### PR TITLE
fix ui scene add batch bug

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: https://github.com/cocos/3d-tasks/issues/14935

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by SantyWang — APPROVED** [review]

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 1]
```diff
@@ -94,9 +95,17 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
...
+    size_t index = 0;
     for (auto* rootNode : _rootNodeArr) {
         walk(rootNode, 1);
         generateBatch(_currEntity, _currDrawInfo);
+
+        auto const scene = rootNode->getScene()->getRenderScene();
+        size_t const count = _batches.size();
+        for (size_t i = index; i < count; i++) {
+            scene->addBatch(_batches.at(i));
+        }
```
Comment (minggo) [inline native/cocos/2d/renderer/Batcher2d.cpp:107]: Can use for loop like
```c++
for (auto* batch : _batches) {
        scene->addBatch(batch);
}
```

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 2]
```diff
@@ -94,9 +95,17 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
...
+        index = count;
```
Comment (minggo) [inline native/cocos/2d/renderer/Batcher2d.cpp:109]: Why assign to count?

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 3]
```diff
@@ -94,9 +95,17 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
 }
 
 void Batcher2d::fillBuffersAndMergeBatches() {
+    size_t index = 0;
     for (auto* rootNode : _rootNodeArr) {
         walk(rootNode, 1);
         generateBatch(_currEntity, _currDrawInfo);
+
+        const auto scene = rootNode->getScene()->getRenderScene();
```
Comment (minggo) [inline native/cocos/2d/renderer/Batcher2d.cpp]: Should use pointer there
```c++
const auto *scene = rootNode->getScene()->getRenderScene();
```

File: native/cocos/2d/renderer/Batcher2d.cpp (hunk 2 above)
Comment (LinYunMo) [reply native/cocos/2d/renderer/Batcher2d.cpp:109]: Cause want to add batch segment generated by the current root to the renderscene of the current root

File: native/cocos/2d/renderer/Batcher2d.cpp [hunk 4]
```diff
@@ -94,9 +95,18 @@ void Batcher2d::syncRootNodesToNative(ccstd::vector<Node*>&& rootNodes) {
 }
 
 void Batcher2d::fillBuffersAndMergeBatches() {
+    size_t index = 0;
     for (auto* rootNode : _rootNodeArr) {
+        // _batches will add by generateBatch
```
Comment (minggo) [inline native/cocos/2d/renderer/Batcher2d.cpp:100]: It is better to add this comment above the code `generateBatch(_currEntity, _currDrawInfo)`.

**Review by minggo — APPROVED** [review]

**Pull request merged**